src/index.ts

```diff
@@ -1 +1,52 @@
-export const checkDangers = ()=>{}
\ No newline at end of file
+/* eslint-disable @typescript-eslint/no-shadow */
+// Provides dev-time typing structure for  `danger` - doesn't affect runtime.
+// https://github.com/danger/danger-js/blob/main/docs/usage/extending-danger.html.md#writing-your-plugin
+import * as TE from "fp-ts/TaskEither";
+import * as E from "fp-ts/Either";
+
+import { pipe } from "fp-ts/function";
+
+import { DangerDSLType } from "../node_modules/danger/distribution/dsl/DangerDSL";
+import { getJiraIdFromPrTitle } from "./utils/titleParser";
+import { printWarningError, renderTickets } from "./dangerRender";
+import { getJiraIssues } from "./jira";
+import { fromJiraToGenericTicket } from "./types";
+import { checkMinLength, matchRegex } from "./utils/validator";
+const MIN_LEN_PR_DESCRIPTION = 10;
+
+declare const danger: DangerDSLType;
+export declare function schedule<T>(asyncFunction: Promise<T>): void;
+
+export const checkDangers = async (): Promise<void> => {
+  const addJiraTicket = pipe(
+    danger.github.pr.title,
+    getJiraIdFromPrTitle,
+    TE.fromOption(() => new Error("Jira ID not found in PR title")),
+    TE.chain(getJiraIssues),
+    TE.map((_) => _.map(fromJiraToGenericTicket)),
```

Comment:
if you use fp-ts array methods your code may be terser ie:

`TE.map(A.map(fromJiraToGenericTicket)),`